File: spring_server/src/main/java/com/semothon/spring_server/room/dto/RoomInfoDto.java
package com.semothon.spring_server.room.dto;

import com.semothon.spring_server.common.service.DateTimeUtil;
import com.semothon.spring_server.room.entity.Room;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class RoomInfoDto {
    private Long roomId;
    private String title;
    private String description;
    private Integer capacity;
    private LocalDateTime createdAt;
    private Integer currentMemberCount;
    private Long chatRoomId;

    public static RoomInfoDto from(Room room){
        return RoomInfoDto.builder()
                .roomId(room.getRoomId())
                .title(room.getTitle())
                .description(room.getDescription())
                .capacity(room.getCapacity())
                .createdAt(DateTimeUtil.convertUTCToKST(room.getCreatedAt()))
                .currentMemberCount(room.getRoomUsers() != null ? room.getRoomUsers().size() : 0)
                .chatRoomId(room.getChatRoom() != null ? room.getChatRoom().getChatRoomId() : null)
                .build();
    }
}

File: flutter_app/lib/dto/room_info_dto.dart
class RoomInfoDto {
  final int roomId;
  final String title;
  final String description;
  final int capacity;
  final DateTime createdAt;
  final int currentMemberCount;
  final int? chatRoomId;
  final List<String> interests;

  RoomInfoDto({
    required this.roomId,
    required this.title,
    required this.description,
    required this.capacity,
    required this.createdAt,
    required this.currentMemberCount,
    required this.chatRoomId,
    required this.interests,
  });

  factory RoomInfoDto.fromJson(Map<String, dynamic> json) {
    return RoomInfoDto(
      roomId: json['roomId'],
      title: json['title'],
      description: json['description'],
      capacity: json['capacity'],
      createdAt: DateTime.parse(json['createdAt']),
      currentMemberCount: json['currentMemberCount'],
      chatRoomId: json['chatRoomId'],
      interests: List<String>.from(json['interests']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'roomId': roomId,
      'title': title,
      'description': description,
      'capacity': capacity,
      'createdAt': createdAt.toIso8601String(),
      'currentMemberCount': currentMemberCount,
      'chatRoomId': chatRoomId,
      'interests': interests,
    };
  }
}
